### PR TITLE
Add Feature attribute to all IShapeTableProvider implementations

### LINKED ISSUE #14103: Configuring instance of `IShapeTableProvider` does not to the IoC container correctly without `[Feature()]` attribute
### Describe the bug

I have the following startup class

```
[Feature("CrestApps.Components.User.Avatars")]
public class AvatarStartup : StartupBase
{
    public override void ConfigureServices(IServiceCollection services)
    {
       //....
        services.AddScoped<IShapeTableProvider, AvatarShapeTableProvider>();
    }
}
```

If I enable the `CrestApps.Components.User.Avatars` feature, the `AvatarShapeTableProvider` is registered correctly. If I disable the feature after that, the ShellContext is released. However, the `AvatarShapeTableProvider` does not get removed from the IoC even when the `AvatarStartup` does not get called again. It's like it gets cached or registered in a global `IoC` container instead of the tenant's container which is very strange.

To solve the problem, I had to explicitly add the `[Feature("CrestApps.Components.User.Avatars")]` to the `AvatarShapeTableProvider` class. 

It is my understanding that `[Feature("...")]` attribute is required on the controller only.

@jtkech have you seen this issue before? It is very strange to have to add `[Feature("...")]` for every implementation of `IShapeTableProvider`. Using `ConfigureServices()` should be all that we have to do.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #14103

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Media/MediaShapes.cs [hunk 1]
```diff
@@ -1,8 +1,11 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.Modules;
 
 namespace OrchardCore.ContentFields.Media
 {
+    [Feature("OrchardCore.ContentFields")]
+    [RequireFeatures("OrchardCore.Media")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Media/MediaShapes.cs:8]: Need to be tried but I think the feature attribute is not needed when related to the module main feature.

But the require feature attribute seems to make sense.

File: src/OrchardCore.Modules/OrchardCore.Contents/Shapes.cs [hunk 2]
```diff
@@ -5,9 +5,11 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
 using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Contents
 {
+    [Feature("OrchardCore.Contents")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Contents/Shapes.cs:12]: Need to be tried but I think the feature attribute is not needed when related to the module main feature.

File: src/OrchardCore.Modules/OrchardCore.Html/Media/MediaShapes.cs [hunk 3]
```diff
@@ -1,7 +1,10 @@
 using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Html.Media
 {
+    [Feature("OrchardCore.Html")]
+    [RequireFeatures("OrchardCore.Media")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Html/Media/MediaShapes.cs:7]: Idem

File: src/OrchardCore.Modules/OrchardCore.Markdown/Media/MediaShapes.cs [hunk 4]
```diff
@@ -1,8 +1,11 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Markdown.Media
 {
+    [Feature("OrchardCore.Markdown")]
+    [RequireFeatures("OrchardCore.Media")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Markdown/Media/MediaShapes.cs:8]: Idem

File: src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs [hunk 5]
```diff
@@ -6,10 +6,12 @@
 using OrchardCore.DisplayManagement.Descriptors;
 using OrchardCore.DisplayManagement.Shapes;
 using OrchardCore.Menu.Models;
+using OrchardCore.Modules;
 using OrchardCore.Mvc.Utilities;
 
 namespace OrchardCore.Menu
 {
+    [Feature("OrchardCore.Menu")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs:14]: Need to be tried but I think the feature attribute is not needed when related to the module main feature.

File: src/OrchardCore.Modules/OrchardCore.Navigation/NavigationShapes.cs [hunk 6]
```diff
@@ -4,10 +4,12 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
 using OrchardCore.DisplayManagement.Shapes;
+using OrchardCore.Modules;
 using OrchardCore.Mvc.Utilities;
 
 namespace OrchardCore.Navigation
 {
+    [Feature("OrchardCore.Navigation")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Navigation/NavigationShapes.cs:12]: idem

File: src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs [hunk 7]
```diff
@@ -13,9 +13,11 @@
 using OrchardCore.DisplayManagement.Html;
 using OrchardCore.DisplayManagement.Implementation;
 using OrchardCore.DisplayManagement.Shapes;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Navigation
 {
+    [Feature("OrchardCore.Navigation")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs:20]: Idem

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/TermShapes.cs [hunk 8]
```diff
@@ -7,12 +7,14 @@
...
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.Modules;
 using OrchardCore.Mvc.Utilities;
 using OrchardCore.Taxonomies.Models;
 using OrchardCore.Taxonomies.ViewModels;
 
 namespace OrchardCore.Taxonomies
 {
+    [Feature("OrchardCore.Taxonomies")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Taxonomies/TermShapes.cs:17]: Idem

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs [hunk 9]
```diff
@@ -1,10 +1,13 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
 using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Modules;
 using OrchardCore.Tenants.ViewModels;
 
 namespace OrchardCore.Tenants.Services;
 
+[Feature("OrchardCore.Tenants")]
+[RequireFeatures("OrchardCore.Tenants", "OrchardCore.Features")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:10]: Idem for the not needed feature attribute.

Then maybe only`[RequireFeatures("OrchardCore.Tenants.FeatureProfiles")]`

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantShapeTableProvider.cs [hunk 10]
```diff
@@ -1,10 +1,12 @@
 using OrchardCore.DisplayManagement;
 using OrchardCore.DisplayManagement.Descriptors;
 using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Modules;
 using OrchardCore.Tenants.ViewModels;
 
 namespace OrchardCore.Tenants.Services;
 
+[Feature("OrchardCore.Tenants")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantShapeTableProvider.cs:9]: Idem, I think not needed when related to the main feature of the module.

File: src/OrchardCore.Modules/OrchardCore.Widgets/ContentCardShapes.cs [hunk 11]
```diff
@@ -1,8 +1,10 @@
 using System;
 using OrchardCore.DisplayManagement.Descriptors;
+using OrchardCore.Modules;
 
 namespace OrchardCore.Widgets
 {
+    [Feature("OrchardCore.Widgets")]
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Widgets/ContentCardShapes.cs:7]: Idem

File: src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs (hunk 5 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Menu/MenuShapes.cs]: Done

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/TermShapes.cs (hunk 8 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Taxonomies/TermShapes.cs]: done

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs (hunk 9 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:9]: This class is registered here
```
    [RequireFeatures("OrchardCore.Tenants", "OrchardCore.Features")]
    public class TenantFeatureProfilesStartup : StartupBase
    {
        public override void ConfigureServices(IServiceCollection services)
        {
            services.AddScoped<IShapeTableProvider, TenantFeatureShapeTableProvider>();
        }
    }
```

so we should only need `[RequireFeatures("OrchardCore.Tenants", "OrchardCore.Features")]`

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantShapeTableProvider.cs (hunk 10 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantShapeTableProvider.cs]: done

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Media/MediaShapes.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.ContentFields/Media/MediaShapes.cs:7]: done

File: src/OrchardCore.Modules/OrchardCore.Contents/Shapes.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Contents/Shapes.cs]: done

File: src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs (hunk 7 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Navigation/PagerShapes.cs]: done

File: src/OrchardCore.Modules/OrchardCore.Navigation/NavigationShapes.cs (hunk 6 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Navigation/NavigationShapes.cs]: done

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs (hunk 9 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:9]: @jtkech are you good with this PR? Can we merge it?

**Comment by MikeAlhayek** [comment]
@jtkech any other changes you have for this PR?

File: src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs (hunk 9 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:9]: @jtkech you may have forgotten about this one.

Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:9]: Oops, will see tomorrow.

> so we should only need [RequireFeatures("OrchardCore.Tenants", "OrchardCore.Features")]

Hmm, maybe `OC.Tenants` is not needed here too if defined in `OC.Tenants` itself.

Don't remember the details, would need to be tried.

Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantFeatureShapeTableProvider.cs:9]: Yes I agree. I'll remove it from the startup and the filter.

**Review by jtkech — APPROVED** [review]

**Pull request merged**